blog/2025-08-26-verifying-bls12-on-ethereum.md

```diff
@@ -0,0 +1,40 @@
+---
+title: "Verifying the `quicknet` beacons on Ethereum"
+description: "At long last!"
+date: 2025-08-26
+author: Julie Bettens
+tags:
+    - Beginner
+---
+Let us explore together the effectivness of EIP-2537 after the Pectra upgrade for onchain BLS signature verification.
+<!-- truncate -->
+
+The League of Entropy has been emitting `quicknet` random beacons for anyone to use for all their public verifiable randomness uses [since 2023](https://docs.drand.love/blog/2023/10/16/quicknet-is-live), and ever since we launched it people have been repeatedly asking us about how to verify drand beacons “onchain”, especially on Ethereum. Today we finally have everything we need to answer that question!
+
+To make drand work, we need to use a mathematical construction known as a *pairing-friendly elliptic curve*. [You can read more about this in our documentation.](https://docs.drand.love/docs/cryptography/#pairing-based-cryptography) Nowadays, there are two main contenders to chose from: BN254 and BLS12-381. BN254 has the advantage of being supported by many blockchain networks because it is a bit older, whereas BLS12-381 offers better performance and higher security. Since our launch in 2020, drand has relied on BLS12-381, although the League of Entropy has also discreetly launched a special `evmnet` beacon in 2024 that uses BN254 to be compatible with more blockchain networks.
```

Comment:
```suggestion
To make drand work, we need to use a mathematical construction known as a *pairing-friendly elliptic curve*. [You can read more about this in our documentation.](https://docs.drand.love/docs/cryptography/#pairing-based-cryptography) Nowadays, there are two main contenders to choose from: BN254 and BLS12-381. BN254 has the advantage of being supported by many blockchain networks because it is a bit older, whereas BLS12-381 offers better performance and higher security. Since our launch in 2020, drand has relied on BLS12-381, although the League of Entropy has also discreetly launched a special `evmnet` beacon in 2024 that uses BN254 to be compatible with more blockchain networks.
```